### DISCUSSION TITLE
Packages with similar goals as lua-widow-control

### DISCUSSION THREAD

**Discussion by hpvd** [gucci-on-fleek/lua-widow-control#6] [General]
Would be nice to promote other package which also enhances LaTeX **automatic typesetting quality** to the next level (like this lua widow control does) within in the doc

In the end, when talking about each other all packages will gain more visibility and all LaTeX documents shines even more...

e.g.:
1. fewerfloatpages
https://www.ctan.org/pkg/fewerfloatpages
https://github.com/FrankMittelbach/fmitex-fewerfloatpages

> LATEX’s float algorithm has the tendency to produce fairly empty float pages, i.e., pages
> containing only floats but with a lot of free space remaining that could easily be filled
> with nearby text. There are good reasons for this behavior; nevertheless, the results
> look unappealing and in many cases documents are unnecessarily enlarged.
> The fewerfloatpages package provides an extended algorithm that improves on this
> behavior **without the need for manual intervention** by the user.

2. the well known microtype
https://www.ctan.org/pkg/microtype
https://github.com/schlcht/microtype/

**Comment by gucci-on-fleek** [comment]
There's also the [linebreaker package](https://github.com/michal-h21/linebreaking) which I find pretty neat. Usually TeX makes 3 attempts to break a paragraph: first without hyphenation, then with hyphenation, then adding `\emergencystretch` to each line. The linebreaker package extends this so that instead of adding the `\emergencystretch` all in one attempt, it progressively adds more and more of the `\emergencystretch`  through multiple passes. This doesn't affect the breaking of "good" paragraphs, but it improves the spacing of any paragraphs that needed to use `\emergencystretch`.
